src/test/scala/com/microsoft/hyperspace/index/rules/RuleUtilsTest.scala

```diff
@@ -143,15 +146,18 @@ class RuleUtilsTest extends HyperspaceRuleSuite with SQLHelper {
               (if (hybridScanDeleteEnabled) "0.99" else "0")) {
             val indexes = RuleUtils
               .getCandidateIndexes(spark, allIndexes, plan)
-            if (expectCandidateIndex) {
-              assert(indexes.length === 1)
+              .sortBy(_.name)
+            if (numExpectedCandidateIndex > 0) {
+              assert(indexes.length === numExpectedCandidateIndex)
               assert(indexes.head.name === "index1")
```

Comment:
Should we just check all the names? `assert(indexes.map(_.name) === (1 to numExpectedCandidateIndexes).map("index" + _))` (didn't test compiling)

Reply:
or it may just be clear to pass the name of expected indexes instead of numExpectedCandidateIndexes: `expectedCandidateIndexNames: Seq[String]`